proposals/collection-expression-better-conversion.md

```diff
@@ -0,0 +1,125 @@
+
+These rules are:
+
+> Given an implicit conversion `C₁` that converts from an expression `E` to a type `T₁`, and an implicit conversion `C₂` that converts from an expression `E` to a type `T₂`, `C₁` is a ***better conversion*** than `C₂` if one of the following holds:
+> **`E` is a *collection expression*, and `C₁` is a ***better collection conversion from expression***, or**
+> * **`E` is not a *collection expression* and one of the following holds:**
+>   * `E` exactly matches `T₁` and `E` does not exactly match `T₂`
+>   * `E` exactly matches both or neither of `T₁` and `T₂`, and `T₁` is a [*better conversion target*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#11646-better-conversion-target) than `T₂`
+> * `E` is a method group, ...
+
+We add a new definition for ***better collection conversion from expression***, as follows:
+
+Given:
+- `E` is a collection expression with element expressions `[EL₁, EL₂, ..., ELₙ]`
+- `T₁` and `T₂` are collection types
+- `T'₁`:
+    - If `T₁` is a generic type, `T₁` with all type parameters unsubstituted
+    - If `T₁` is an array type, an unsubstituted array type
+    - Otherwise, `T₁`
+- `T'₂` is the same as `T'₁`, except for using `T₂` instead of `T₁`
+- `E₁` is the element type of `T₁`
+- `E₂` is the element type of `T₂`
+- `CE₁ᵢ` are the series of conversions from `ELᵢ` to `E₁`
+- `CE₂ᵢ` are the series of conversions from `ELᵢ` to `E₂`
+
+`T'₁` is at least as good as `T'₂` if there is an identity conversion between them.  
+`T'₁` is better than `T'₂` if:
+- `T'₁` is ReadOnlySpan<T>, and `T'₂` is not ReadOnlySpan<T>
+- `T'₂` is Span<T>, and `T'₂` is an *array_or_array_interface*
+- `T'₂` is implicitly convertible to `T'₁`, and `T'₁` is not implicitly convertible to `T'₂`.
+  
+Otherwise, `T'₁` neither as good as nor better than `T₂`.
+
+If `E₁` and `E₂` have an identity conversion, then the element conversions are as good as each other. Otherwise, the element conversions to `E₁` are better than the element conversions to `E₂` if:
+- For every `ELᵢ`, `CE₁ᵢ` is at least as good as `CE₂ᵢ`, and
+- There is at least one i where `CE₁ᵢ` is better than `CE₂ᵢ`
+  
+Otherwise, neither set of element conversions is better than the other, and they are also not as good as each other.  
+Conversion comparisons are made using better conversion from expression if `ELᵢ` is not a spread expression. If `ELᵢ` is a spread expression, we use better conversion from the element type of the spread collection to `E₁` or `E₂`, respectively.
+
+`T₁` is a ***better collection conversion from expression*** than `T₂` if:
+- `T'₁` is better than `T'₂`, and the element conversions to `E₁` are at least as good as the element conversions to `E₂`, or
+- `T'₁` is at least as good as `T'₂`, and the element conversions to `E₁` are better than the element conversions to `E₂`.
```

Comment:
```suggestion
- `T'₁` is the same as `T'₂`, and the element conversions to `E₁` are at least as good as than the element conversions to `E₂`, and at least one of the element conversions to `E₁` is better than the element conversion to `E₂`.
```

Reply:
Not holding an opinion about which way, but I did notice there was something satisfying about the symmetry of "A is better and B is at least as good, or A is at least as good and B is better."

Reply:
I used the wording I did specifically here.